File: VB/SeriesNonVisualWrapper.vb
Imports DevExpress.Xpf.Charts
Imports System.Windows.Media

Namespace CheckBoxesInLegendExample_UnboundMode

    Friend Class SeriesNonVisualWrapper

        Public Property Series As Series

        Public Property MarkerBrush As Brush

        Public Property MarkerLineBrush As Brush

        ' This property is bound to legend marker template.
        Public Property MarkerLineStyle As LineStyle

        Public Sub New(ByVal series As Series, ByVal markerBrush As Brush)
            Me.Series = series
            Me.MarkerBrush = markerBrush
            MarkerLineStyle = New LineStyle(1)
            MarkerLineBrush = markerBrush
        End Sub
    End Class
End Namespace

File: CS/SeriesNonVisualWrapper.cs
using DevExpress.Xpf.Charts;
using System.Windows.Media;

namespace CheckBoxesInLegendExample_UnboundMode {
    
    class SeriesNonVisualWrapper {

        public Series Series { 
            get; 
            set; 
        }
        public Brush MarkerBrush {
            get;
            set;
        }
        public Brush MarkerLineBrush {
            get;
            set;
        }

        // This property is bound to legend marker template.
        public DevExpress.Xpf.Charts.LineStyle MarkerLineStyle { 
            get; 
            set; 
        }

        public SeriesNonVisualWrapper(Series series, Brush markerBrush) {
            Series = series;
            MarkerBrush = markerBrush;
            MarkerLineStyle = new DevExpress.Xpf.Charts.LineStyle(1);
            MarkerLineBrush = markerBrush;
        }
    }
}